codec-http/src/main/java/io/netty5/handler/codec/http/HttpObjectDecoder.java

```diff
@@ -809,173 +907,221 @@ private void splitHeader(AppendableCharSequence sb) {
+        return LATIN_WHITESPACE[b + 128];
+    }
+
+    private static int findNonWhitespace(byte[] sb, int offset, int end) {
+        for (int result = offset; result < end; ++result) {
+            byte c = sb[result];
+            if (!isWhitespace(c)) {
                 return result;
             } else if (!isOWS(c)) {
                 // Only OWS is supported for whitespace
                 throw new IllegalArgumentException("Invalid separator, only a single space or horizontal tab allowed," +
-                                                   " but received a '" + c + "' (0x" + Integer.toHexString(c) + ')');
+                        " but received a '" + c + "' (0x" + Integer.toHexString(c) + ')');
             }
         }
-        return sb.length();
+        return end;
     }
 
-    private static int findEndOfString(AppendableCharSequence sb) {
-        for (int result = sb.length() - 1; result > 0; --result) {
-            if (!isOWS(sb.charAtUnsafe(result))) {
+    private static int findEndOfString(byte[] sb, int start, int end) {
+        for (int result = end - 1; result > start; --result) {
+            if (!isOWS(sb[result])) {
                 return result + 1;
             }
         }
         return 0;
     }
 
-    private static boolean isOWS(char ch) {
-        return ch == ' ' || ch == (char) 0x09;
+    private static boolean isOWS(byte ch) {
+        return ch == ' ' || ch == 0x09;
     }
 
-    private static class HeaderParser implements ByteProcessor {
-        private final AppendableCharSequence seq;
-        private final int maxLength;
+    private static class HeaderParser {
+        protected final Buffer seq;
+        protected final int maxLength;
         int size;
 
-        HeaderParser(AppendableCharSequence seq, int maxLength) {
+        HeaderParser(Buffer seq, int maxLength) {
             this.seq = seq;
             this.maxLength = maxLength;
         }
 
-        public AppendableCharSequence parse(Buffer buffer) {
-            final int oldSize = size;
-            seq.reset();
-            int i = buffer.openCursor().process(this);
-            if (i == -1) {
-                size = oldSize;
+        public Buffer parse(Buffer buffer) {
+            final int readableBytes = buffer.readableBytes();
+            final int readerIndex = buffer.readerOffset();
+            final int maxBodySize = maxLength - size;
+            assert maxBodySize >= 0;
+            // adding 2 to account for both CR (if present) and LF
+            // don't remove 2L: it's key to cover maxLength = Integer.MAX_VALUE
+            final long maxBodySizeWithCRLF = maxBodySize + 2L;
+            final int toProcess = (int) Math.min(maxBodySizeWithCRLF, readableBytes);
+            final int toIndexExclusive = readerIndex + toProcess;
+            assert toIndexExclusive >= readerIndex;
+            int toLf = buffer.bytesBefore(HttpConstants.LF);
+            final int indexOfLf = readerIndex + toLf;
```

Comment:
Is it using the swat index of?

Reply:
Yes, it is: https://github.com/netty/netty/blob/main/buffer/src/main/java/io/netty5/buffer/bytebuffer/NioBuffer.java#L385-L415